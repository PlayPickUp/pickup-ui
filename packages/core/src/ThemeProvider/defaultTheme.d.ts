import { DefaultTheme } from "../../../../types";
export declare const defaultTheme: DefaultTheme;
//# sourceMappingURL=defaultTheme.d.ts.map
