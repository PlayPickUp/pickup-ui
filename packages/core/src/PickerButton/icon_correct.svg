<svg width="14" height="20" viewBox="0 0 14 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M2 10.3506L5.6958 13.7412L11.81 6.57227" stroke="#25DA95" stroke-width="2.5" stroke-linejoin="round"/>
</svg>
