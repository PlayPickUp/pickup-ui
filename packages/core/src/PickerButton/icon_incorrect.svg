<svg width="14" height="20" viewBox="0 0 14 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M3 6L11 14" stroke="#FF1244" stroke-width="2.5"/>
<path d="M11 6L3 14" stroke="#FF1244" stroke-width="2.5"/>
</svg>
