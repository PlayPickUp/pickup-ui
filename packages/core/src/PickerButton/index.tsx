import React from "react";
import { createUseStyles } from "react-jss";
import classNames from "classnames";
import color from "color";

import bolt from "./bolt.svg";

import { DefaultTheme, PickerButtonProps } from "../types";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  "@keyframes growRight": {
    from: { width: 0 },
    to: { width: (props) => `${props.result}%` },
  },
  root: {
    display: "block",
    position: "relative",
    width: "100%",
    height: "auto",
    minHeight: 38,
    padding: theme.spacing.base,
    backgroundColor: theme.colors.grey.light,
    fontSize: 14,
    color: theme.colors.black,
    border: `1px solid ${theme.colors.grey.base}`,
    borderRadius: 5,
    textAlign: "center",
    transition: `background-color 75ms ease`,
    "&:hover": {
      backgroundColor: color(theme.colors.grey.light).darken(0.05).toString(),
    },
  },
  result: {
    backgroundColor: theme.colors.white,
    "&:hover": {
      backgroundColor: theme.colors.white,
    },
  },
  resultPicked: {
    backgroundColor: theme.colors.white,
    border: `1px solid ${theme.colors.purple.base}`,
    "&:hover": {
      backgroundColor: theme.colors.white,
    },
  },
  textContainer: {
    display: "flex",
    position: "relative",
    width: "100%",
    flexFlow: "row nowrap",
    justifyContent: "space-between",
    alignItems: "center",
    padding: `0 ${theme.spacing.base * 2}px`,
    zIndex: 2,
  },
  resultContainer: {
    display: "flex",
    position: "relative",
    flexFlow: "row nowrap",
    justifyContent: "flex-end",
    alignItems: "center",
  },
  bolt: {
    display: "none",
  },
  isPick: {
    display: "block",
    "& img": {
      position: "relative",
      width: 11,
      height: "auto",
      marginRight: theme.spacing.base * 3,
    },
  },
  bar: {
    display: "block",
    position: "absolute",
    top: 4,
    left: 4,
    width: "calc(100% - 8px)",
    height: "calc(100% - 8px)",
    borderRadius: 2,
    zIndex: 1,
    "& > div": {
      display: "block",
      position: "absolute",
      top: 0,
      left: 0,
      width: 0,
      height: "100%",
      backgroundColor: theme.colors.purple.light,
    },
  },
  barActive: {
    "& > div": {
      animation: "$growRight 1s cubic-bezier(0.280, 0.520, 0.630, 0.990) 1",
      width: (props) => `${props.result}%`,
    },
  },
}));

const PickerButton: React.FC<PickerButtonProps> = ({
  className,
  displayText,
  onClick,
  style,
  showResult = false,
  result,
  isPick,
}) => {
  const classes = useStyles({ result });

  return (
    <button
      data-testid="picker-button"
      className={classNames({
        [classes.root]: true,
        [classes.result]: showResult,
        [classes.resultPicked]: isPick,
        className: className,
      })}
      onClick={onClick}
      style={style}
    >
      <div
        className={classNames({
          [classes.bar]: true,
          [classes.barActive]: showResult,
        })}
      >
        <div />
      </div>
      {!showResult ? (
        <div style={{ zIndex: 2, position: "relative" }}>{displayText}</div>
      ) : (
        <div className={classes.textContainer}>
          <div>{displayText}</div>
          <div className={classes.resultContainer}>
            <div
              className={classNames({
                [classes.bolt]: true,
                [classes.isPick]: isPick,
              })}
            >
              <img src={bolt} aria-hidden role="presentation" />
            </div>
            <div>{result}%</div>
          </div>
        </div>
      )}
    </button>
  );
};

export default PickerButton;
