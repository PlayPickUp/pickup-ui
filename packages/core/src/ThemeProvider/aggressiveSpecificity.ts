/* This is based on the unsupported jss-increase-specificity plugin at https://github.com/iamstarkov/jss-increase-specificity
JSS's original createGenerateId has been adapted here to use logic from that plugin. /*

// From the original plugin:
/**
 * use :not(#\20), :not(.\20) and :not(\20) instead of generating unlikely
 * appearing ids…
 * — twitter.com/subzey/status/829050478721896448
 * Rationale: \20 is a css escape for U+0020 Space, and neither classname,
 * nor id, nor tagname can contain a space
 * — twitter.com/subzey/status/829051085885153280
 */

import { Rule, CreateGenerateId } from "jss";
import { JssStyleSheet } from "../types";

var selector = ":not(#\\20)";
const repeat = 3;
const maxRules = 1e10;

export const increaseSpecificity: CreateGenerateId = (options = {}) => {
  let ruleCounter = 0;
  console.log({ options });
  return (rule: Rule, sheet?: JssStyleSheet): string => {
    ruleCounter += 1;
    if (ruleCounter > maxRules) {
      console.warn(
        false,
        `[JSS] You might have a memory leak. Rule counter is at ${ruleCounter}.`
      );
    }
    const jssId = String(sheet.options.jss.id) ?? "";
    const prefix = sheet.options.classNamePrefix ?? "";

    let suffix = "";
    for (let i = 0; i < repeat; i++) {
      suffix += selector;
    }

    return `${prefix + rule.key}${
      jssId ? `-${jssId}` : ""
    }-${ruleCounter}${suffix}`;
  };
};
