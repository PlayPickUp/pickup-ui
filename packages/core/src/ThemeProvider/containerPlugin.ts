/*
 *  Adds container params to all selectors
 *  Example: 'h1' becomes 'body #PickUpUI h1'
 */

import { Plugin } from "jss";

export function containUI(withReset: boolean, container?: string): Plugin {
  const prefix = container ? container : "body #PickUpUI ";

  // 'rule' is left untyped because the Rule import from jss is not recognizing all child properties
  function onProcessRule(rule): Plugin["onProcessRule"] {
    const parent = rule.options.parent;
    if (
      rule.type !== "style" ||
      rule.selectorText.includes(prefix) ||
      (parent && parent.type === "keyframes") ||
      (withReset && parent && parent.key === "@global")
    )
      return;

    if (rule.selectorText.includes(",")) {
      const allSelectors = rule.selectorText.split(",");
      const transformedSelectors = allSelectors.map((selector: string) =>
        selector.includes(prefix) ? selector : prefix + selector
      );
      rule.selectorText = transformedSelectors.join(",");
    } else {
      rule.selectorText = prefix + rule.selectorText;
    }
  }
  return { onProcessRule };
}
