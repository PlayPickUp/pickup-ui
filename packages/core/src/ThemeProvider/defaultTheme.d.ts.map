{"version":3,"file":"defaultTheme.d.ts","sourceRoot":"","sources":["defaultTheme.ts"],"names":[],"mappings":"AAAA,OAAO,EAMN,YAAY,EAWZ,MAAM,mBAAmB,CAAC;AAqJ3B,eAAO,MAAM,YAAY,EAAE,YAU1B,CAAC"}