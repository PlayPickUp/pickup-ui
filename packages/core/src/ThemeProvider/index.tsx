import React from "react";
import { jss, JssProvider, ThemeProvider as JSSThemeProvider } from "react-jss";
import { containUI } from "./containerPlugin";
import merge from "lodash/merge";

import { defaultTheme } from "./defaultTheme";

import { ThemeProviderProps } from "../types";
import GlobalsAndReset from "./GlobalsAndReset";

function generateRandom(digitCount) {
  return Math.floor(
      Math.pow(10, digitCount - 1) +
      Math.random() * 9 * Math.pow(10, digitCount - 1)
  );
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  withReset = true,
  theme,
  ...rest
}) => {
  jss.use(containUI(withReset));
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;

  const random = generateRandom(5)

  return (
    <div id="PickUpUI">
      <JssProvider jss={jss} classNamePrefix={`PU${random}--`}>
        <JSSThemeProvider theme={mergedTheme} {...rest}>
          <GlobalsAndReset>{children}</GlobalsAndReset>
        </JSSThemeProvider>
      </JssProvider>
    </div>
  );
};

export default ThemeProvider;
