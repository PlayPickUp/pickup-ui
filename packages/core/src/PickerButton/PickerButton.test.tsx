import React from "react";
import { fireEvent, render } from "@testing-library/react";
import jssSerializer from 'jss-snapshot-serializer';
expect.addSnapshotSerializer(jssSerializer);
import ThemeProvider from "../ThemeProvider";
import PickerButton from "./index";

const handleClick = jest.fn();

test("Renders and matches snapshot", () => {
  const { container } = render(
    <ThemeProvider>
      <PickerButton
        result={74}
        displayText="Absolutely"
        isPick={false}
        onClick={handleClick}
      />
      <PickerButton
        onClick={handleClick}
        displayText="Definitely Not"
        style={{ marginBottom: 10 }}
        showResult={true}
        result={33.33}
        isIncorrect
      />
      <PickerButton
        onClick={handleClick}
        displayText="Absolutely"
        style={{ marginBottom: 10 }}
        result={56}
        showResult={true}
        isCorrect
      />
      <PickerButton
        onClick={handleClick}
        displayText="Maybe?"
        style={{ marginBottom: 10 }}
        result={33.33}
        showResult={true}
      />
    </ThemeProvider>
  );
  expect(container).toMatchSnapshot();
});

test("Display text renders correctly", () => {
  const { getByText } = render(
    <ThemeProvider>
      <PickerButton
        isPick={false}
        result={85}
        displayText="Absolutely"
        onClick={handleClick}
      />
    </ThemeProvider>
  );
  expect(getByText("Absolutely")).toBeTruthy();
});

test("On click fires correctly", () => {
  const { getByTestId } = render(
    <ThemeProvider>
      <PickerButton
        result={34}
        displayText="Absolutey"
        onClick={handleClick}
        isPick={false}
      />
    </ThemeProvider>
  );
  fireEvent.click(getByTestId("picker-button"));
  expect(handleClick).toHaveBeenCalledTimes(1);
});

test("Display of results text renders correctly", () => {
  const { getByText } = render(
    <ThemeProvider>
      <PickerButton
        showResult={true}
        isPick={true}
        result={85}
        displayText="Absolutely"
        onClick={handleClick}
      />
    </ThemeProvider>
  );
  expect(getByText("Absolutely")).toBeTruthy();
  expect(getByText("85%")).toBeTruthy();
});
