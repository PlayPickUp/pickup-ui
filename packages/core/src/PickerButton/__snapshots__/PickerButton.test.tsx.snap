// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders and matches snapshot 1`] = `
<div>
  <button
    class="root-0-2-2"
    data-testid="picker-button"
  >
    <div
      class="nonPickedBar-0-2-10"
    >
      <div />
    </div>
    <div
      style="z-index: 2; position: relative;"
    >
      Absolutely
    </div>
  </button>
  <button
    class="root-0-2-2 result-0-2-3 isIncorrect-0-2-12"
    data-testid="picker-button"
    style="margin-bottom: 10px;"
  >
    <div
      class="nonPickedBar-0-2-10 barActive-0-2-15 barActive-d3-0-2-19 barIncorrect-0-2-14"
    >
      <div />
    </div>
    <div
      class="textContainer-0-2-5"
    >
      <div>
        Definitely Not
      </div>
      <div
        class="resultContainer-0-2-6"
      >
        <div
          class="bolt-0-2-7 hasIcon-0-2-8"
        >
          <img
            aria-hidden="true"
            role="presentation"
            src="test-file-stub"
          />
        </div>
        <div>
          33.33
          %
        </div>
      </div>
    </div>
  </button>
  <button
    class="root-0-2-2 result-0-2-3 isCorrect-0-2-11"
    data-testid="picker-button"
    style="margin-bottom: 10px;"
  >
    <div
      class="nonPickedBar-0-2-10 barActive-0-2-15 barActive-d5-0-2-21 barCorrect-0-2-13"
    >
      <div />
    </div>
    <div
      class="textContainer-0-2-5"
    >
      <div>
        Absolutely
      </div>
      <div
        class="resultContainer-0-2-6"
      >
        <div
          class="bolt-0-2-7 hasIcon-0-2-8"
        >
          <img
            aria-hidden="true"
            role="presentation"
            src="test-file-stub"
          />
        </div>
        <div>
          56
          %
        </div>
      </div>
    </div>
  </button>
  <button
    class="root-0-2-2 result-0-2-3"
    data-testid="picker-button"
    style="margin-bottom: 10px;"
  >
    <div
      class="nonPickedBar-0-2-10 barActive-0-2-15 barActive-d7-0-2-23"
    >
      <div />
    </div>
    <div
      class="textContainer-0-2-5"
    >
      <div>
        Maybe?
      </div>
      <div
        class="resultContainer-0-2-6"
      >
        <div
          class="bolt-0-2-7"
        >
          <img
            aria-hidden="true"
            role="presentation"
          />
        </div>
        <div>
          33.33
          %
        </div>
      </div>
    </div>
  </button>
</div>
`;
